azero/contracts/token/lib.rs

```diff
@@ -227,11 +219,16 @@ pub mod token {
     impl Mintable for Token {
         #[ink(message)]
         fn mint(&mut self, to: AccountId, value: u128) -> Result<(), PSP22Error> {
-            self.ensure_minter_burner()?;
+            self.ensure_minter()?;
             let events = self.data.mint(to, value)?;
             self.emit_events(events);
             Ok(())
         }
+
+        #[ink(message)]
+        fn minter(&self) -> AccountId {
+            self.minter
+        }
     }
 
     impl Burnable for Token {
```

Comment:
One thing that I'm not sure about: do we even want anyone but minter to be able to burn tokens?
Seems like it can cause a user error that causes the bridge to lose its invariant of having 1:1 correspondence between minted and locked tokens.

Reply:
Yes, good point. I think maybe indeed only the "admin" should be able to burn.

Reply:
I've always thought that the setup would be so that `addPair` on the MOST contract creates the instance of PSP22 token, setting itself as the "owner"/admin of that token.

Reply:
Yes, it can be done this way too, but I think the current approach is OK -- the only serious downside is that it requires more manual steps.